### ISSUE TITLE
Examples upgrade

### ISSUE THREAD

**Issue by andrii-bodnar** [lingui/js-lingui#1393]
The [examples](https://github.com/lingui/js-lingui/tree/main/examples) are extremely outdated and need to be upgraded

- [ ] [create-react-app](https://github.com/lingui/js-lingui/tree/main/examples/create-react-app)
- [ ] [js](https://github.com/lingui/js-lingui/tree/main/examples/js)
- [ ] [next-js](https://github.com/lingui/js-lingui/tree/main/examples/next-js)

**Comment by andrii-bodnar** [comment]
> Also we should create 2 examples for next-js (with and without SWC)

We have without SWC - [js-lingui/examples/next-js](https://github.com/lingui/js-lingui/tree/main/examples/next-js), with SWC - [swc-plugin/examples/nextjs-13](https://github.com/lingui/swc-plugin/tree/main/examples/nextjs-13), isn't it?

**Comment by thekip** [comment]
Correct, but it is hard to discover when they exist in two different repos, and it's hard to understand a difference between them. 

I would suggest: 

create two folders with distinctive name: 
`examples/next-js-with-babel`
`examples/next-js-with-swc`

And either move nextjs example from SWC repo to here, or create in `examples/next-js-with-swc` just a readme file with reference to swc-plugin repo example. 

It's quite handy when you have all examples in one place, like there https://github.com/vercel/next.js/tree/canary/examples

Also nextjs example in the main repo is broken. Here i posted an explanation https://github.com/lingui/js-lingui/issues/1265

**Comment by andrii-bodnar** [comment]
Yeah, it's better to have all the examples in one place and additionally we should consider adding a small section with the links so users will be able to easily find all these examples.

**Comment by andrii-bodnar** [comment]
https://github.com/vercel/next.js/tree/canary/examples/with-lingui

**Comment by bafxyz** [comment]
I would be happy to see at least one working example with nodejs, typescript and extract parser. So far could not find anything, or lingui is not intended to work with nodejs without hacks?

**Comment by thekip** [comment]
@bafxyz if you accepting to use babel or swc for precompiling you backend code, everything should work without any hacks. but bear in mind that lingui is made mostly for frontend code and some solutions might be overengenireed for backend code.

**Comment by thekip** [comment]
the nodejs example is too wide. it all depends on the frameworks you use. For example, nestjs and express would be completely different examples.

**Comment by gpessa** [comment]
> I would be happy to see at least one working example with nodejs, typescript and extract parser. So far could not find anything, or lingui is not intended to work with nodejs without hacks?

Are we sure that with this example the server side page is generated?
It use `useEffect` which only called on the client side.

**Comment by cbratschi** [comment]
Is there a Next.js App Router example?

**Comment by bafxyz** [comment]
> Is there a Next.js App Router example?

Try this one https://github.com/sultan99/next.js-v13

**Comment by cbratschi** [comment]
@bafxyz thanks, trying it right now. Just failed with adapting the SWC example code.

**Comment by cbratschi** [comment]
Well, if I use a client component, the following error occurs:

```
./node_modules/babel-plugin-macros/node_modules/cosmiconfig/dist/readFile.js:9:0
Module not found: Can't resolve 'fs'
```

And in case of a server component, t is not defined as a function.

Is this supposed to support both cases? At least the middleware checks both cases and the used provider should also support it but the lingui code fails.

**Comment by thekip** [comment]
@cbratschi that means you're trying to use Macro without actual transpiling Macro. To use macro you have to use either [babel-plugin-macros](https://github.com/kentcdodds/babel-plugin-macros) or [@lingui/swc-plugin](https://github.com/lingui/swc-plugin)

**Comment by Collinbrown95** [comment]
Would an example with [RSpack](https://www.rspack.dev/) be useful?

I made a separate issue for this (https://github.com/lingui/js-lingui/issues/1749), but this issue might be more appropriate (in which case I can close https://github.com/lingui/js-lingui/issues/1749 in favour of this one).

**Comment by sohamnandi77** [comment]
@thekip , please add some turborepo + vite + react examples as well, monorepo docs are not much clear

**Comment by n-ce** [comment]
Please add Vite + Typescript example.

**Comment by abdonrd** [comment]
Are there any plans to create an example of a monorepo?
The Docs are almost empty: https://lingui.dev/guides/monorepo
Thanks!

**Comment by timofei-iatsenko** [comment]
I'm thinking of writing a tutorial on the monorepo topic, but im still not there. The main problem with monorepos - there many ways to organize it and hard to cover all possible scenarios in one article or example

**Comment by abdonrd** [comment]
Ok, thanks @timofei-iatsenko!

And any examples/recommendations for a package that is going to be published in an npm registry?

**Comment by abdonrd** [comment]
Ok, thanks @timofei-iatsenko!

And any examples/recommendations for a package that is going to be published in an npm registry?

✅ I have a Next.js v15 (App Router) working here:
https://github.com/abdonrd/next-lingui-monorepo-example/commit/2c2c731efc6ae26fe3187319119bbc017b4c61fb

🏗️ But it still doesn't work when I try to translate another package from the monorepo:
https://github.com/abdonrd/next-lingui-monorepo-example/commit/cf87ab57274f0249341582afd991158be5469369
